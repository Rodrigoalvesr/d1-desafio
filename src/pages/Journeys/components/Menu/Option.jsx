import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Icon from '@Components/Icon';
import Badge from '@Components/Badge';
import getStatus from '@/common/getStatus';
import { makeStyles } from '@material-ui/core/styles';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    marginBottom: 10,
    userSelect: 'none',
  },
  content: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    cursor: 'pointer',
    '&:hover': {
      '& $name': {
        color: theme.palette.primary.main,
      },
      '& $badge:not(.selected)': {
        background: '#EBEEF6',
      },
    },
  },
  icon: {
    width: 16,
    height: 16,
    marginRight: 10,
  },
  badge: {},
  name: {
    fontSize: 13,
    '&.selected': {
      fontWeight: 'bold',
      color: '#117EFF',
    },
  },
}));

function Option({ id, name, quantity, selected, onClick }) {
  const classes = useStyles();
  const status = React.useMemo(() => getStatus(id), [id]);

  return (
    <div className={classes.root}>
      <Icon className={classes.icon} name={status.icon} color={status.color} />

      <div className={classes.content} onClick={onClick}>
        <Typography className={classnames(classes.name, { selected })}>
          {name}
        </Typography>

        <Badge selected={selected} value={quantity} className={classes.badge} />
      </div>
    </div>
  );
}

Option.defaultProps = {
  selected: false,
};

Option.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  selected: PropTypes.bool,
};

export default Option;
