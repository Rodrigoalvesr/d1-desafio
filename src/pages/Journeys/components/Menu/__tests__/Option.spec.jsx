import React from 'react';
import renderer from 'react-test-renderer';
import Option from '../Option';

describe('Option', () => {
  it('Renders correctly', () => {
    const component = renderer
      .create(
        <Option id={1} name="Em execução" quantity={2} onClick={jest.fn} />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
