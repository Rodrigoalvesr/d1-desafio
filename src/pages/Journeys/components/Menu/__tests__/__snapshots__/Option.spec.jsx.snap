// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <svg
    className="makeStyles-root-6 makeStyles-root-7 makeStyles-icon-3"
  >
    paper-plane.svg
  </svg>
  <div
    className="makeStyles-content-2"
    onClick={[Function]}
  >
    <p
      className="MuiTypography-root makeStyles-name-5 MuiTypography-body1"
    >
      Em execução
    </p>
    <div
      className="makeStyles-root-8 makeStyles-badge-4"
    >
      <p
        className="MuiTypography-root makeStyles-text-9 MuiTypography-body1"
      >
        2
      </p>
    </div>
  </div>
</div>
`;
